src/api/spec/controllers/webui/patchinfo_controller_spec.rb

```diff
@@ -0,0 +1,269 @@
+require 'rails_helper'
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the Patchinfo methods
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::PatchinfoController, vcr: true do
+  let(:user) { create(:user, login: 'macario') }
+  let(:other_user) { create(:confirmed_user, login: 'gilberto')}
+  let(:other_package) { create(:package, project: user.home_project, name: 'other_package') }
+  let(:patchinfo_package) do
+    patchinfo = Patchinfo.new.create_patchinfo(user.home_project_name, nil)
+    Package.get_by_project_and_name(user.home_project_name, patchinfo[:targetpackage], use_source: false)
+  end
+  let(:do_proper_post_save) do
+    post :save, project: user.home_project_name,
+                package: patchinfo_package.name,
+                summary: 'long enough summary is ok',
+                description: 'long enough description is also ok' * 5,
+                issueid: [769484],
+                issuetracker: ['bgo'],
+                issuesum: [],
+                issueurl: ['https://bugzilla.gnome.org/show_bug.cgi?id=769484'],
+                category: 'recommended',
+                rating: 'low',
+                packager: user.login
+  end
+
+  after do
+    Package.destroy_all
+  end
+
+  describe 'POST #new_patchinfo' do
+    before do
+      other_user
+      login user
+    end
+
+    context 'without permission to create the patchinfo package' do
+      before do
+        post :new_patchinfo, project: other_user.home_project
+      end
+
+      it { expect(response).to redirect_to(project_show_path(other_user.home_project)) }
+      it { expect(flash[:error]).to eq "No permission to create packages" }
+    end
+
+    context 'when it fails to create the patchinfo package' do
+      before do
+        Patchinfo.any_instance.stubs(:create_patchinfo).returns(false)
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to redirect_to(project_show_path(user.home_project)) }
+      it { expect(flash[:error]).to eq "Error creating patchinfo" }
+    end
+
+    context 'when the patchinfo package file is not found' do
+      before do
+        Package.any_instance.stubs(:patchinfo).returns(nil)
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to redirect_to(package_show_path(project: user.home_project, package: 'patchinfo')) }
+      it { expect(flash[:error]).to eq("Patchinfo not found for #{user.home_project.name}") }
+    end
+
+    context 'when is successfull creating the patchinfo package' do
+      before do
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to have_http_status(:success) }
+      it { expect(assigns(:binaries)).to be_a Array }
```

Comment:
That's very loose. I am not sure if it makes sense to test that it is an array. Either test for having the correct binaries or remove the test.